Assets/Scripts/SS3D/Systems/Electricity/FuelPowerGenerator.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Coimbra.Services.Events;
+using Coimbra.Services.PlayerLoopEvents;
+using FishNet.Object;
+using FishNet.Object.Synchronizing;
+using SS3D.Core;
+using SS3D.Data.Enums;
+using SS3D.Systems.Audio;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using AudioType = SS3D.Systems.Audio.AudioType;
+
+namespace System.Electricity
+{
+    public class FuelPowerGenerator : BasicElectricDevice, IPowerProducer
+    {
+        [SerializeField]
+        private float _powerProduction = 10f;
+
+        [SerializeField]
+        private SkinnedMeshRenderer _skinnedMeshRenderer;
+
+        private const string OnBlendShapeName = "On";
+
+        private const string OutputBlendShapeName = "Output";
+
+        private const string LowFuel = "LowFuel";
+        public float PowerProduction => _powerProduction;
+
+        private Quaternion _initialRotation; // Rotation of the generator at rest.
+
+        private Vector3 _directionOfShake; // In which direction the generator shake.
```

Comment:
Maybe the vibration stuff should be in a separate file so it can be easily referenced/used by multiple machines?

Reply:
I can make that autonomous yeah.